test/e2e/specs/editor/various/autocomplete-and-mentions.spec.js

```diff
@@ -0,0 +1,411 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const userList = [
+	{
+		username: 'testuser',
+		firstName: 'Jane',
+		lastName: 'Doe',
+		password: 'iLoVeE2EtEsTs',
+	},
+	{
+		username: 'yourfather',
+		firstName: 'Darth',
+		lastName: 'Vader',
+		password: 'padme123',
+	},
+	{
+		username: 'mockingjay',
+		firstName: 'Katniss',
+		lastName: 'Everdeen',
+		password: 'district12forlyfe',
+	},
+	{
+		username: 'ringbearer',
+		firstName: 'Frodo',
+		lastName: 'Baggins',
+		password: 'ep1cburgl@r',
+	},
+	{
+		username: 'thebetterhobbit',
+		firstName: 'Bilbo',
+		lastName: 'Baggins',
+		password: 'lostwithouts@m',
+	},
+	{
+		username: 'makeitso',
+		firstName: 'Jean-Luc',
+		lastName: 'Picard',
+		password: 'engagE!1',
+	},
+	{
+		username: 'buddytheelf',
+		firstName: 'Buddy',
+		lastName: 'Elf',
+		password: 'sm1lingsmyfavorite',
+	},
+];
+test.describe( 'Autocomplete', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await Promise.all(
+			userList.map( ( user ) =>
+				requestUtils.createUser( {
+					email: `${ user.username }@example.com`,
+					...user,
+				} )
+			)
+		);
+		await requestUtils.activatePlugin( 'gutenberg-test-autocompleter' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllUsers();
+		await requestUtils.deactivatePlugin( 'gutenberg-test-autocompleter' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { editor } ) => {
+		await editor.publishPost();
+	} );
+
+	[
+		[ 'User Mention', 'mention' ],
+		[ 'Custom Completer', 'option' ],
+	].forEach( ( completerAndOptionType ) => {
+		const [ completer, type ] = completerAndOptionType;
+		test( `${ completer }: should insert ${ type }`, async ( {
+			page,
+			editor,
+		} ) => {
+			// Set up test data for each case
+			const testData = {};
+			if ( type === 'mention' ) {
+				testData.triggerString = 'I am @da';
+				testData.optionText = 'Darth Vader';
+				testData.snapshot = `<!-- wp:paragraph -->
+<p>I am @yourfather.</p>
+<!-- /wp:paragraph -->`;
+			} else if ( type === 'option' ) {
+				testData.triggerString = 'I like ~s';
+				testData.optionText = '🍓 Strawberry';
+				testData.snapshot = `<!-- wp:paragraph -->
+<p>I like 🍓.</p>
+<!-- /wp:paragraph -->`;
+			}
+
+			await page.click( 'role=button[name="Add default block"i]' );
+			await page.keyboard.type( testData.triggerString );
+			await expect(
+				page.locator( 'role=option', { hasText: testData.optionText } )
```

Comment:
Could and should we explicitly specify the `name` for the options?

```suggestion
				page.locator( `role=option[name="Darth Vader yourfather"]` )
```

Reply:
Done! 1ead803